nullaway/src/test/java/com/uber/nullaway/jspecify/ArrayTests.java

```diff
@@ -608,6 +608,25 @@ public void mismatchedIndexUse() {
         .doTest();
   }
 
+  @Test
+  public void typeUseAndDeclarationLegacyAnnotationOnArray() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jetbrains.annotations.Nullable;",
+            "class Test {",
+            "  @Nullable Object[] foo1 = null;",
```

Comment:
In my table, I wrote that if an annotation is both type use and declaration and placed on the elements, we do _not_ have a `@Nullable` array ref.  This test seems to show the opposite.  Was this deliberate?  Do you think we should change the table?  It may well be this is the best thing to do.

Reply:
From what I saw the declaration behaviors seem to take precedence for annotations which are both. I think we can probably simply change the table, or I can look into inverting the precedence

Reply:
This is tricky.  One the one hand, just reading the above code, if you're used to JSpecify mode, I think you would be surprised that the array reference itself becomes `@Nullable`.  On the other hand, if it were a pure declaration annotation, then this surprising behavior remains.  So I think it's reasonable to say that if you have an annotation that is both declaration _and_ type use, written in a legal position for a delcaration annotation, the declaration nature of the annotation "wins".  I'll go ahead and change the table accordingly.